src/setup-gcloud.ts

```diff
@@ -100,7 +101,26 @@ export async function run(): Promise<void> {
         credsPath = path.join(githubWorkspace, uniqueName);
       }
 
-      const serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
+      let serviceAccountKeyObj;
+      // if explicit SA passed, use that
+      if (serviceAccountKey) {
+        serviceAccountKeyObj = parseServiceAccountKey(serviceAccountKey);
+      } else if (process.env.GOOGLE_GHA_CREDS_PATH) {
+        const cPath = await fs.readFile(
+          process.env.GOOGLE_GHA_CREDS_PATH,
+          'utf8',
+        );
+        serviceAccountKeyObj = parseServiceAccountKey(cPath);
```

Comment:
I don't think this will work, because the export from `auth` could be a WIF cred file, which isn't a "service account key"

Reply:
I refactored the logic a bit based on what we discussed yesterday and added some test cases for more scenarios. I also added a warning to help users migrating that export_creds is not necessary when using setup-gcloud in conjunction with auth.